File: java/src/pattern/strategy/adventure/Character.java
package pattern.strategy.adventure;

public abstract class Character
{
    protected String name;
    protected Weapon weapon;

    public Character()
    {
	setName("Der Namenlose");
	setWeapon(new Fists());
    }

    public void setName(String name)
    {
	this.name = name;
    }

    public void setWeapon(Weapon weapon)
    {
	this.weapon = weapon;
    }

    public String toString()
    {
	return name;
    }

    public void fight()
    {
	System.out.println(this + " " + weapon.fight() + " mit " + weapon);
    }

    public void fight(Character character)
    {
	System.out.println(this + " " + weapon.fight() + " " + character + " mit " + weapon);
    }
}

File: php/src/pattern/strategy/adventure/Character.php
<?php

namespace pattern\strategy\adventure;

abstract class Character
{
    protected $_name;
    protected $_weapon;

    public function __construct()
    {
	$this->setName("Der Namenlose");
	$this->setWeapon(new Fists());
    }

    public function setName($name)
    {
	$this->_name = $name;
    }

    public function setWeapon(Weapon $weapon)
    {
	$this->_weapon = $weapon;
    }

    public function __toString()
    {
	return $this->_name;
    }

    public function fight(Character $character = null)
    {
	($character) ? $target = (string) $character : $target = '';
	echo "{$this} {$this->_weapon->fight()} {$target} mit {$this->_weapon}\n";
    }

}
